core/src/main/java/io/grpc/internal/ServerImplBuilder.java

```diff
@@ -98,7 +100,8 @@
    */
   public interface ClientTransportServersBuilder {
     InternalServer buildClientTransportServers(
-        List<? extends ServerStreamTracer.Factory> streamTracerFactories);
+        List<? extends ServerStreamTracer.Factory> streamTracerFactories,
+        ChannelLogger channelLogger);
```

Comment:
I agree that some of the changes might be very unnecessary looking at it now, thanks for the opinion.
Because I didn't want the `ChannelLoggerImpl` to be visible outside of the package and was also hesitating to duplicate [toJavaLogLevel](https://github.com/grpc/grpc-java/blob/e5ab4d743da0e9268905369a83f2c4faeee77d4e/core/src/main/java/io/grpc/internal/ChannelLoggerImpl.java#L97) somewhere else, maybe I can move it to a more common place, like io.grpc.internal.GrpcUtil?

Reply:
It wouldn't be that big of a deal to expose `toJavaLogLevel()`; all the classes are in `io.grpc.internal` and `toJavaLogLevel()` is static. I'd probably just make the method package-private.

But copying isn't that bad in this case; the mapping is trivial, unlikely to change, and not a big deal if client vs server happen to get out-of-sync. In either case, "don't spend much time on it" is the main goal.